addons/website_quote/models/sale_order.py

```diff
@@ -22,9 +22,11 @@ class SaleOrderLine(models.Model):
     @api.onchange('product_id')
     def product_id_change(self):
         domain = super(SaleOrderLine, self).product_id_change()
-        if self.order_id.template_id:
-            self.name = next((quote_line.name for quote_line in self.order_id.template_id.quote_line if
-                             quote_line.product_id.id == self.product_id.id), self.name)
+        if self.product_id and self.order_id.template_id:
+            for quote_line in self.order_id.template_id.quote_line:
+                if quote_line.product_id and quote_line.product_id.id == self.product_id.id:
```

Comment:
Small note: quote_line.product_id == self.product_id is sufficient (you can compare record sets and you know self.product_id is set due to first if)

Reply:
Changed